### PR TITLE
 ipc: align stream.h with kernel

### PR DESCRIPTION
**Pull request by mrajwa**
This change is to align sof_ipc_stream_params with its
    kernel equivalent. The change is needed in both kernel
    and FW because FW needs to use host_period_bytes which
    where previously used to inform host about period IRQ.
    Therefore now we introduce new member for it period_irq
    and let host_period_bytes keep the proper value.
    
Corresponding Kernel PR is here https://github.com/thesofproject/linux/pull/1037

    Signed-off-by: Marcin Rajwa <marcin.rajwa@linux.intel.com>

### CODE REVIEW

File: src/include/ipc/stream.h [hunk 1]
```diff
@@ -92,8 +92,10 @@ struct sof_ipc_stream_params {
 
 	/* for notifying host period has completed - 0 means no period IRQ */
 	uint32_t host_period_bytes;
+	uint16_t period_irq;
 
-	uint32_t reserved[2];
+	uint16_t not_used;
```
Comment (tlauda) [inline src/include/ipc/stream.h]: It will look nicer if we merge them together.

Comment (mrajwa) [reply src/include/ipc/stream.h]: Sure I can define reserved as `uint16_t ` and make it `uint32_t reserved[5];` then remove `not_used` competely.

File: src/audio/host.c [hunk 2]
```diff
@@ -108,7 +108,7 @@ static void host_update_position(struct comp_dev *dev, uint32_t bytes)
 		hd->local_pos = 0;
 
 	/* NO_IRQ mode if host_period_size == 0 */
-	if (dev->params.host_period_bytes != 0) {
+	if (dev->params.period_irq != 0) {
```
Comment (plbossart) [inline src/audio/host.c]: fix comment above?

Comment (mrajwa) [reply src/audio/host.c]: @plbossart, ahh I overlooked it. Thanks for spotting it. I have updated both sides, can we merge it together now?

File: src/include/ipc/stream.h [hunk 3]
```diff
@@ -92,8 +92,9 @@ struct sof_ipc_stream_params {
 
 	/* for notifying host period has completed - 0 means no period IRQ */
 	uint32_t host_period_bytes;
+	uint16_t period_irq;
 
-	uint32_t reserved[2];
+	uint16_t reserved[3];
```
Comment (plbossart) [inline src/include/ipc/stream.h]: you also have a comment that needs to be fixed line 93 and in general it helps when you add more details on a boolean. in other places we use no_irq, here it's period_irq so 50% of people will get it wrong :-)

File: src/audio/host.c [hunk 4]
```diff
@@ -107,8 +107,8 @@ static void host_update_position(struct comp_dev *dev, uint32_t bytes)
 	if (hd->local_pos >= hd->host_size)
 		hd->local_pos = 0;
 
-	/* NO_IRQ mode if host_period_size == 0 */
-	if (dev->params.host_period_bytes != 0) {
+	/* NO_IRQ mode if period_irq == 0 */
+	if (dev->params.period_irq != 0) {
```
Comment (keyonjie) [inline src/audio/host.c]: you will break skl- platforms, for those platforms, we need period_irq = 1, but you haven't set it anywhere.

I suggest to use no_period_irq(by default with value 0, suitable for skl-, and set it to 1 here for skl+ platforms).

File: src/audio/host.c [hunk 5]
```diff
@@ -107,8 +107,8 @@ static void host_update_position(struct comp_dev *dev, uint32_t bytes)
 	if (hd->local_pos >= hd->host_size)
 		hd->local_pos = 0;
 
-	/* NO_IRQ mode if host_period_size == 0 */
-	if (dev->params.host_period_bytes != 0) {
+	/* NO_IRQ mode if no_period_irq == 1 */
+	if (dev->params.no_period_irq != 0) {
```
Comment (keyonjie) [inline src/audio/host.c]: if (!dev->params.no_period_irq) {

File: src/include/ipc/stream.h [hunk 6]
```diff
@@ -90,10 +90,10 @@ struct sof_ipc_stream_params {
 	uint16_t sample_valid_bytes;
 	uint16_t sample_container_bytes;
 
-	/* for notifying host period has completed - 0 means no period IRQ */
 	uint32_t host_period_bytes;
+	uint16_t no_period_irq;
```
Comment (keyonjie) [inline src/include/ipc/stream.h]: please align comments with kernel PR also.

File: src/include/ipc/stream.h (hunk 3 above)
Comment (lgirdwood) [reply src/include/ipc/stream.h]: @mrajwa is this now resolved ? We can merge if resolved.

Comment (mrajwa) [reply src/include/ipc/stream.h]: @lgirdwood yes I think we can merge both sides simultaneously.

**Pull request merged**